### PR TITLE
finish rocketmq cpp 4.x test(client cluster filter server pull)

### PR DESCRIPTION
**Pull request by Zintoki**
## Brief changelog

Added the test content for the cpp4.x client

## Verifying this change

Content:

 - [x] Test some Settings of 4.x client message, pushconsumer, pullconsumer and producer
 - [x] Test Clust, single producer, multiple pushconsumer reception
 - [x] Test the Tag, and find the relevant information content through the tag
 - [x] Test the pullconsumer (message ACK, and message NACK)
 - [x] Test four kinds of messages , delay messages, normal messages, order messages, and transaction messages

截屏2023-10-20 20 29 42

### CODE REVIEW

**Comment by lizhanhui** [comment]
A problem to address is, how to trigger and run these E2E tests for each pull request to ensure they do not break anything.

**Comment by Zintoki** [comment]
> A problem to address is, how to trigger and run these E2E tests for each pull request to ensure they do not break anything.

OK, I'll try to write a shell script that helps automate build(download the cpp library and compile) and execute these E2E tests.

**Comment by lizhanhui** [comment]
> > A problem to address is, how to trigger and run these E2E tests for each pull request to ensure they do not break anything.
> 
> OK, I'll try to write a shell script that helps automate build(download the cpp library and compile) and execute these E2E tests.

Can it be part of the [required status check](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/collaborating-on-repositories-with-code-quality-features/troubleshooting-required-status-checks) for the pull request?

**Comment by Zintoki** [comment]
> > > A problem to address is, how to trigger and run these E2E tests for each pull request to ensure they do not break anything.
> > 
> > 
> > OK, I'll try to write a shell script that helps automate build(download the cpp library and compile) and execute these E2E tests.
> 
> Can it be part of the [required status check](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/collaborating-on-repositories-with-code-quality-features/troubleshooting-required-status-checks) for the pull request?

Yes, but sudo privileges may be required. The "privileged: true" option may be required to raise the privilege level when writing workflow so that cpp libraries can be quickly installed for cpp4.x e2e testing

File: cpp/bin/run.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+# limitations under the License.
+
+
+# cd project base dir to compile mqadmin utils for other language e2e test using
+cd ../common &&  mvn -Prelease -DskipTests clean package -U
+# set env for mqadmin (use source to set linux env variables in current shell)
+cd ../rocketmq-admintools && source bin/env.sh
+# run cpp e2e test case
+cd ../cpp/rocketmq-client-cpp-tests/cpp4.x
+cmake . -B build && cmake --build build
```
Comment (fakerpawnno1) [inline cpp/bin/run.sh:25]: to make it run ，maybe should install library required first in this file

**Review by fakerpawnno1 — COMMENTED** [review]
LGTM

**Pull request merged**